cypress/integration/test.spec.js

```diff
@@ -0,0 +1,153 @@
+const headShowPlayIconSelector = '.info .header svg';
+const headShowGoToShowButtonSelector = '.info .header button';
+const showsSelector = '.MuiGrid-container .MuiGrid-item:last-child div:last-child .MuiGrid-container .MuiGrid-item';
+const rusFlagButton = '.MuiGrid-container .MuiGrid-item:first-child ul li button:first-child';
+const enFlagButton = '.MuiGrid-container .MuiGrid-item:first-child ul li button:last-child';
+const playerSelector = '.MuiGrid-container div:last-child audio';
+const menuTabsSelector = '[aria-label="Menu tabs"] button';
+const tableShowsSelector = 'tbody tr';
+
+const sideBarPages = ['/shows', '/library', '/account', '/'];
+const getSideBarLinkSelector = page => `a[href="${page}"]`;
+const flagsButtons = [
+  {
+    selector: rusFlagButton,
+    text: latestShowsRu.latestShowsTitle
+  },
+  {
+    selector: enFlagButton,
+    text: latestShowsEn.latestShowsTitle
+  }
+];
+
+describe('Test Latest Shows', () => {
+  let firstShow = {};
+  let popularShows = [];
+
+  before(async () => {
+    popularShows = await getData(popularShowsUrl);
+    [firstShow] = popularShows;
+  });
+
+  it('home page must have head latest show', () => {
+    cy.visit('/');
+    cy.get(headShowTitleSelector).should('have.text', firstShow.title);
+    cy.get(headShowImageSelector).should('have.attr', 'src', firstShow.image);
+  });
+
+  it('home page must have latest show list', () => {
+    cy.visit('/');
+    cy.get(showsSelector).should('have.length', popularShows.length - 1);
+  });
+
+  it('head show play icon must open player', () => {
+    cy.visit('/');
+    cy.get(headShowPlayIconSelector).click();
+    cy.get(playerSelector).should('exist');
+  });
+
+  it('move to show page on head show button click', () => {
+    cy.visit('/');
+    cy.get(headShowGoToShowButtonSelector).click();
+    cy.url().should('include', `/shows/${firstShow.id}`);
+  });
+});
+
+
+describe('Test Side Bar', () => {
+  it(`side bar's nav button must redirect to different page`, () => {
+    cy.visit('/')
+    sideBarPages.forEach(page => {
+      cy.get(getSideBarLinkSelector(page)).first().click();
+      cy.location('pathname').should('eq', `${page}`);
+    });
+  });
+
+  it('flag buttons must change language of static data', () => {
+    cy.visit('/');
+    flagsButtons.forEach(flag => {
+      cy.get(flag.selector).click();
+      cy.findByText(flag.text).should('exist');
+    });
+  });
+});
+
+
+describe('Test Shows', () => {
+  let shows = [];
+  let firstShow = {};
+
+  before(async () => {
+    shows = await getData(showsUrl);
+    [firstShow] = shows;
+  });
+
+  it('show page must have show list', () => {
+    cy.visit('/shows');
+    cy.get(showsSelector).should('have.length', shows.length);
+  });
+
+  it('search input must find valid show', () => {
+    cy.visit('/shows');
+    cy.get('input').type(firstShow.title);
+    cy.get(showsSelector).should('have.length', 1);
+  });
+
+  // it('play sound buttons must run player', () => {
```

Comment:
Not sure it's a correct way of checking. Once the play button was pressed and the player appeared it will be present on the page, no need to test it with `forEach`. You can press the play button in the first episode, after -> pressing the play button on the other episode -> and checking that sources are different and the player shows the` Pause` icon (means the track is still running).